code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java

```diff
@@ -436,19 +438,30 @@ public static void injectBalancerLabelsAndHealthcheck(Map<Object, Object> launch
     private static boolean doesLBHaveDrainSupport(Map<Object, Object> launchConfig) {
         if(launchConfig.get(InstanceConstants.FIELD_IMAGE_UUID) != null) {
             String imageUuid = (String)launchConfig.get(InstanceConstants.FIELD_IMAGE_UUID);
-            DockerImage img = DockerImage.parse(imageUuid);
-            String[] imageParts = img.getFullName().split(":");
-            if (imageParts.length <= 1) {
+            Pair<String, String> instanceImage = getImageAndVersion(imageUuid.toLowerCase());
+            if (instanceImage.getLeft().isEmpty() || instanceImage.getRight().isEmpty()) {
                 return false;
             }
-            if (!imageParts[0].equals("rancher/lb-service-haproxy")) {
+            Pair<String, String> defaultImage = getImageAndVersion(LB_IMAGE_UUID.get().toLowerCase());
+            if (!defaultImage.getLeft().equalsIgnoreCase(instanceImage.getLeft())) {
```

Comment:
looks like equalsignorecase is redundant as we are lowercasing both images anyway